### ISSUE TITLE
Use QMP to directly attach ceph volumes

### ISSUE THREAD

**Issue by markdryan** [01org/ciao#549]
ciao-launcher can attach ceph volumes to VM instances in two ways:

1. When the instance boots.  It does this my specifying a device and drive on the qemu-system-x86-64 command line.  This is the most efficient method.
2. It can also live attach a ceph volume to a running instance using QMP.  This is done by first mapping the ceph volume to a device on the host and then attaching that device, e.g., /dev/rbd0, to the running instance.

It's confusing having two different methods of attaching volumes to instances and it makes the code more complicated as we need to ensure that the two approaches are compatible, e.g., a hotplugged volume is available after reboot and a volume attached at boot can be detached while the instance is running.

The reason the code is like this currently is that I could not figure out how to attach a rbd volume directly using QMP, i.e., without first mapping it to a /dev/rbd0 device.  It may be still possible to do this.  We just need to spend some time loooking at the qemu code to see if we can figure it out.  If it's not possible we can close this ticket.  If it is possible we can speed up live attaches and simplify the launcher code.

**Comment by markdryan** [comment]
I did some investigation and it seems that this feature is not yet supported in QEMU, even in 2.8.0.  However, 2.7 and 2.8 did add Blockdev_Add support for new drivers such as nbd, so lets hope rbd gets added soon.  In the meantime we're just going to have to live with this extra code complexity.

**Comment by tpepper** [comment]
Should we put in a wishlist "bug" at https://bugs.launchpad.net/qemu/ so it's on their list too?  If we had time we could probably even make the patch...

**Comment by markdryan** [comment]
@tpepper Agreed.  The good news is that this bug becomes a lot less critical for us with our chosen migration model.  Most of the problems this issue creates relates to restarting stopped instances.  And since we're no longer going to be doing that on launcher the problem mostly goes away along with the horrid code that manages this stuff.  The only remaining issue will be that hotplugged volumes will be slightly less performant than volumes attached at boot, until the instance is migrated.
